polkadot/xcm/src/v3/multiasset.rs

```diff
@@ -517,9 +516,16 @@ impl MaxEncodedLen for MultiAssets {
 }
 
 impl Decode for MultiAssets {
-	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, parity_scale_codec::Error> {
-		Self::from_sorted_and_deduplicated(Vec::<MultiAsset>::decode(input)?)
-			.map_err(|()| "Out of order".into())
+	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
```

Comment:
Without knowing the context here; I suppose we dont need to track this globally via environmental?  
I hope its enough just in combination with the overall instruction + nesting limit, but its good to double-check.

Reply:
There's no way to nest `MultiAsset`s, so it's okay to not track anything globally.
Each individual `MultiAssets` being decoded will be able to have at most `MAX_ITEMS_IN_MULTIASSETS = 20` assets.
The limit set on instructions will make sure you don't add a lot of these.